include/ByteBuffer.hpp

```diff
@@ -407,6 +407,14 @@ class ByteBuffer {
     int32_t capacity() {
         return m_capacity;
     }
+
+    bool operator!=(const ByteBuffer& other) {
+        bool isEqual = (m_capacity == other.m_capacity);
```

Comment:
I can see two different meanings of buffer equality, one is that the two buffers are exactly the same in terms of structure, the other is that the content of the two buffers are the same.

For option 1, you might want to consider checking all member variables, not just the capacity.

For option 2, a better check is to make sure the `m_limit` are the same and that the `m_buffer[0-m_limit]` are the same.

You can also optimize it by first checking to see if `m_buffer` are pointing to the same address.